File: TimberCruisingApp/Screens/ModeSelectionScreen.swift
// Top-level mode picker. Two large vertically-stacked buttons:
//   • Tree Measurement — direct measurement tools without the
//     project/stratum/cruise-design overhead.
//   • Timber Cruising  — full standard cruising workflow. Tree
//     Measurement is also reachable from inside this mode.

import SwiftUI

public struct ModeSelectionScreen: View {

    @EnvironmentObject private var environment: AppEnvironment
    @EnvironmentObject private var settings: AppSettings
    @EnvironmentObject private var history: QuickMeasureHistory

    public init() {}

    public var body: some View {
        NavigationStack {
            VStack(spacing: ForestixSpace.xl) {
                Spacer()

                VStack(alignment: .center, spacing: ForestixSpace.xxs) {
                    Text("FORESTIX")
                        .font(.system(size: 14, weight: .semibold))
                        .tracking(2.4)
                        .foregroundStyle(ForestixPalette.primary)
                    Text("Choose a workflow")
                        .font(ForestixType.title)
                        .foregroundStyle(ForestixPalette.textPrimary)
                }

                VStack(spacing: ForestixSpace.md) {
                    NavigationLink {
                        TreeMeasurementHubScreen()
                    } label: {
                        ModeButton(
                            title: "Tree Measurement",
                            subtitle: "Direct DBH · height · crown · distance · sampling plots",
                            systemImage: "ruler")
                    }
                    .buttonStyle(.plain)
                    .accessibilityIdentifier("mode.treeMeasurement")

                    NavigationLink {
                        TimberCruisingHubScreen()
                    } label: {
                        ModeButton(
                            title: "Timber Cruising",
                            subtitle: "Standard cruise: projects · strata · plots · stand stats",
                            systemImage: "tree")
                    }
                    .buttonStyle(.plain)
                    .accessibilityIdentifier("mode.timberCruising")
                }
                .padding(.horizontal, ForestixSpace.md)

                Spacer()
                Spacer()
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(ForestixPalette.canvas.ignoresSafeArea())
            #if os(iOS)
            .navigationBarHidden(true)
            #endif
        }
    }
}

private struct ModeButton: View {
    let title: String
    let subtitle: String
    let systemImage: String

    var body: some View {
        HStack(spacing: ForestixSpace.md) {
            ZStack {
                RoundedRectangle(cornerRadius: ForestixRadius.control,
                                 style: .continuous)
                    .fill(ForestixPalette.primaryMuted)
                    .frame(width: 64, height: 64)
                Image(systemName: systemImage)
                    .font(.system(size: 28, weight: .medium))
                    .foregroundStyle(ForestixPalette.primary)
            }
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.system(size: 22, weight: .semibold))
                    .foregroundStyle(ForestixPalette.textPrimary)
                Text(subtitle)
                    .font(ForestixType.caption)
                    .foregroundStyle(ForestixPalette.textSecondary)
                    .lineLimit(2)
                    .fixedSize(horizontal: false, vertical: true)
            }
            Spacer(minLength: 0)
            Image(systemName: "chevron.forward")
                .font(.system(size: 16, weight: .semibold))
                .foregroundStyle(ForestixPalette.textTertiary)
        }
        .padding(ForestixSpace.md)
        .frame(maxWidth: .infinity, minHeight: 120, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: ForestixRadius.card,
                             style: .continuous)
                .fill(ForestixPalette.surface))
        .overlay(
            RoundedRectangle(cornerRadius: ForestixRadius.card,
                             style: .continuous)
                .stroke(ForestixPalette.primary.opacity(0.18),
                        lineWidth: 0.75))
        .shadow(color: Color.black.opacity(0.05),
                radius: 8, x: 0, y: 2)
    }
}

File: android/app/src/main/java/com/hcjeong/forestix/ui/screens/ModeSelectionScreen.kt
// Top-level mode picker — port of iOS ModeSelectionScreen. Two large
// vertically-stacked cards: Tree Measurement (direct tools) and Timber
// Cruising (full standard workflow, also contains Tree Measurement).

package com.hcjeong.forestix.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Straighten
import androidx.compose.material.icons.outlined.Forest
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavController
import com.hcjeong.forestix.ui.Routes
import com.hcjeong.forestix.ui.clickableNoRipple
import com.hcjeong.forestix.ui.theme.Forestix
import com.hcjeong.forestix.ui.theme.ForestixRadius
import com.hcjeong.forestix.ui.theme.ForestixSpace

@Composable
fun ModeSelectionScreen(nav: NavController) {
    val colors = Forestix.colors
    val type = Forestix.type
    Box(
        Modifier
            .fillMaxSize()
            .background(colors.canvas),
        contentAlignment = Alignment.Center,
    ) {
        Column(
            Modifier.fillMaxSize(),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Spacer(Modifier.weight(1f))
            Text(
                "FORESTIX",
                style = type.sectionHead.copy(fontSize = 14.sp, letterSpacing = 2.4.sp, fontWeight = FontWeight.SemiBold),
                color = colors.primary,
            )
            Spacer(Modifier.size(ForestixSpace.xxs))
            Text("Choose a workflow", style = type.title, color = colors.textPrimary)
            Spacer(Modifier.size(ForestixSpace.xl))

            Column(
                Modifier
                    .fillMaxWidth()
                    .padding(horizontal = ForestixSpace.md),
                verticalArrangement = Arrangement.spacedBy(ForestixSpace.md),
            ) {
                ModeButton(
                    title = "Tree Measurement",
                    subtitle = "Direct DBH \u00B7 height \u00B7 crown \u00B7 distance \u00B7 sampling plots",
                    icon = Icons.Filled.Straighten,
                ) { nav.navigate(Routes.TREE_HUB) }
                ModeButton(
                    title = "Timber Cruising",
                    subtitle = "Standard cruise: projects \u00B7 strata \u00B7 plots \u00B7 stand stats",
                    icon = Icons.Outlined.Forest,
                ) { nav.navigate(Routes.TIMBER_HUB) }
            }
            Spacer(Modifier.weight(2f))
        }
    }
}

@Composable
private fun ModeButton(
    title: String,
    subtitle: String,
    icon: ImageVector,
    onClick: () -> Unit,
) {
    val colors = Forestix.colors
    val type = Forestix.type
    Surface(
        color = colors.surface,
        shape = ForestixRadius.card,
        modifier = Modifier
            .fillMaxWidth()
            .heightIn(min = 120.dp)
            .clip(ForestixRadius.card)
            .border(0.75.dp, colors.primary.copy(alpha = 0.18f), ForestixRadius.card)
            .clickableNoRipple(onClick),
    ) {
        Row(
            Modifier.padding(ForestixSpace.md),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(ForestixSpace.md),
        ) {
            Box(
                Modifier
                    .size(64.dp)
                    .clip(RoundedCornerShape(ForestixRadius.controlDp))
                    .background(colors.primaryMuted),
                contentAlignment = Alignment.Center,
            ) {
                Icon(icon, contentDescription = null, tint = colors.primary, modifier = Modifier.size(28.dp))
            }
            Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
                Text(title, style = type.title.copy(fontSize = 22.sp, fontWeight = FontWeight.SemiBold), color = colors.textPrimary)
                Text(subtitle, style = type.caption, color = colors.textSecondary, textAlign = TextAlign.Start)
            }
            Icon(
                Icons.AutoMirrored.Filled.KeyboardArrowRight,
                contentDescription = null,
                tint = colors.textTertiary,
                modifier = Modifier.size(16.dp),
            )
        }
    }
}
